webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,167 @@
+      "Modif. Type",
+      "Test Description",
+      "FR on Evaluation Set",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    await waitFor(() => {
+      expectedColumnHeaders.forEach((name, index) => {
+        expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should have default sort as desc for colum header 'FR on Evaluation/Training set'", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() =>
+      expect(
+        screen.getAllByRole("columnheader")[4].getAttribute("aria-sort")
+      ).toBe("descending")
+    );
+  });
+
+  it("should modify the last Column 'Failure Rate' and tooltips of it if the toggle is changed to train/eval", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    // before toggle changes
+    expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
+      "FR on Evaluation Set"
+    );
+    expect(
+      screen.getByLabelText("Failure Rate on Evaluation Set")
+    ).toBeInTheDocument();
+    // after toggle changes
+    fireEvent.click(screen.getByRole("button", { pressed: false }));
+    await waitFor(() => {
+      expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
+        "FR on Training Set"
+      );
+      expect(
+        screen.getByLabelText("Failure Rate on Training Set")
+      ).toBeInTheDocument();
+    });
+  });
+});
+
+describe("PerturbationTestingPreview without failure rate for training set", () => {
+  const handlers = [getPerturbationResponseWithoutTrainFailureRate];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+  it("should not display value if failure rate for a datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    waitFor(() => {
+      // --% percentage is expected to be shown if failure rate for one of the datasplit is unavailable
+      expect(screen.getByText("--%")).toBeVisible();
+      // should have a success color if the failure rate is not present for the datasplit
+      expect(screen.getByText("--%")).toHaveStyle("color: rgb(46, 125, 50)");
+    });
+  });
+});
+
+describe("PerturbationTestingPreview with Failure response", () => {
+  const handlers = [getPerturbationResponseWithFailureResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+  it("should be expected to display error message if API fails", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    waitFor(() => {
+      // expected error message
+      expect(
+        screen.queryByText(
+          "Something went wrong fetching behavioral testing summary"
+        )
+      ).toBeTruthy();
```

Comment:
That's equivalent to
```suggestion
      screen.getByText(
        "Something went wrong fetching behavioral testing summary"
      );
```

Reply:
right "getByText" is more appropriate here

Reply:
And `toBeTruthy()` is unnecessary.